[ENHANCEMENTS] Tutorial Submission Screen, Updated Styles & Highlight.js

Dan6erbond opened the pull request:
# Overview

Add `tutorials/submit` route for tutorial submission with the ToastUI WYSWIG markdown editor and using Highlight.js styles.

## Tutorial Submission Screen

A skeleton for the tutorial submission screen now allows you to use the ToastUI markdown editor with live preview and scroll sync as well as dropdowns for category, language and framework; these need to be implemented in the SDK in order to fetch them from the remote.

## Highlight.js Styles

The Atom One Dark theme higlight.js is being used with some additional custom styles to offer better code syntax highlighting in posts and tutorials.

## Updated App Style

The default `text-align: center` has been removed, and replaced on elements and selectors where required in order to retain the regular LTR flow.

## Other

Emoji support was added to Remarkable - emojis such as `:)` and `:heart:` will be converted on the fly.

Dan6erbond committed:
Add remarkable emoji support

Dan6erbond committed:
Update formatting

Dan6erbond committed:
Update styles

Remove default `text-align: center` and add targeted.

Dan6erbond committed:
Add highlight.js styles

Use Atom One Dark theme.

Dan6erbond committed:
Setup tutorial submission view

Dan6erbond committed:
Update routes

Doemuu reviewed (changes requested):
<h2>General</h2>
By removing <code>text-align: center</code> The headers signin text misaligns. fix this alignment. The Create Tutorial button in Tutorials.vue leads to <code>/create-tutorial</code>. Fix routerlink.
<h2>Design</h2>
Make title more appealing to used design. Filter Selectors need to have less padding, maybe add a title to them, make them fit into the used design. Make the editor fit the page. At least down to the footer.

Dan6erbond committed:
Fix header sign in button alignment and router link

Update tutorial heading color and alignment.

Dan6erbond committed:
Update submission screen styling and components

 - Add input for title and submit button.
 - Update styling of tagging system as toolbar aligned to right.
 - Make editor 550px high.

Dan6erbond committed:
Update styling

Right-align submission title and update toolbar styling.

Dan6erbond committed:
Merge branch 'master' into issues/23
